### PR TITLE
JBEHAVE-1550 Fix examples tables for filtered by meta given stories

### PR DESCRIPTION
**Pull request by Tatsiana-Luchonak**
Cherry-picked from https://github.com/valfirst/jbehave-core/commit/7a6e9a9c3371997c6325844d64be93f7c2dbbb89

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/model/GivenStories.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public List<String> getPaths() {
 
     public boolean requireParameters() {
         for (GivenStory story : stories) {
-            if (story.hasAnchor()) {
+            if (story.hasAnchorExamples() ){
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/GivenStories.java]:
```suggestion
            if (story.hasAnchorExamples()){
```

File: jbehave-core/src/main/java/org/jbehave/core/model/GivenStory.java [hunk 2]
```diff
@@ -54,6 +54,10 @@ public boolean hasAnchorParameters() {
         return hasAnchor() && anchor.matches(PARAMETERS_REGEX);
     }
 
+    public boolean hasAnchorExamples() {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/GivenStory.java]:
```suggestion
    public boolean hasAnchorWithExamplesReference() {
```

File: jbehave-core/src/test/java/org/jbehave/core/parsers/RegexStoryParserBehaviour.java [hunk 3]
```diff
@@ -149,8 +149,11 @@ void shouldParseStoryWithGivenStoriesWithAnchorParameters() {
                 wholeStory, storyPath);
         assertThat(story.getPath(), equalTo(storyPath));
         assertThat(story.getGivenStories().getStories().size(), equalTo(1));
-        GivenStory givenStory = story.getGivenStories().getStories().get(0);
+        GivenStories givenStories = story.getGivenStories();
+        assertThat(givenStories.requireParameters(), equalTo(false));
+        GivenStory givenStory = givenStories.getStories().get(0);
         assertThat(givenStory.hasAnchorParameters(), equalTo(true));
+        assertThat(givenStory.hasAnchorExamples(), equalTo(false));
```
Comment (valfirst) [inline jbehave-core/src/test/java/org/jbehave/core/parsers/RegexStoryParserBehaviour.java]: I think it makes sense to add the positive test as well (or update any existing one if present)

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by uarlouski — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Comment by valfirst** [comment]
Pulled with thanks!

**Pull request closed without merging**